csv import fails when  parse an empty string as a date

## Guidelines

 `apoc.import.csv`  fails when  parse an empty string as a date
But empty string and int process fine the same procedure .

## Expected Behavior (Mandatory)

Empty or null date in the csv should import like empty string and empty int 

## Actual Behavior (Mandatory)

Failed to invoke procedure `apoc.import.csv`: Caused by: org.neo4j.exceptions.TemporalParseException: Text cannot be parsed to a Date
""
 ^
## How to Reproduce the Problem


### Simple Dataset (where it's possibile)
pass.csv

id:ID,:LABEL,str:STRING,int:INT,date:DATE
1,lab,hello,1,2020-01-01
2,lab,world,2,2020-01-01
3,lab,,,2020-01-01

fail.csv

id:ID,:LABEL,str:STRING,int:INT,date:DATE
1,lab,hello,1,2020-01-01
2,lab,world,2,2020-01-01
3,lab,,,



### Steps (Mandatory)

1 .CALL apoc.import.csv([{fileName: 'pass.csv', labels:[]}], [], {nullValues: ['']})`
works fine in ignoring the blank strings skipping the attribute creation.

However,
2 . `CALL apoc.import.csv([{fileName: 'fail.csv', labels:[]}], [], {nullValues: ['']})`
will fail with

Failed to invoke procedure `apoc.import.csv`: Caused by: org.neo4j.exceptions.TemporalParseException: Text cannot be parsed to a Date
""
 ^

## Screenshots (where it's possibile)

## Specifications (Mandatory)

4.4.4

### Versions

  - OS:ubuntu
  - Neo4j:4.4.4
  - Neo4j-Apoc:apoc-4.4.0.1-all.jar

--- a/core/src/main/java/apoc/load/Mapping.java
+++ b/core/src/main/java/apoc/load/Mapping.java
@@ -3,6 +3,7 @@ package apoc.load;
 import apoc.load.util.LoadCsvConfig;
 import apoc.meta.Meta;
 import apoc.util.Util;
+import org.apache.commons.lang3.StringUtils;
 import org.neo4j.values.storable.DateTimeValue;
 import org.neo4j.values.storable.DateValue;
 import org.neo4j.values.storable.DurationValue;
@@ -70,7 +71,9 @@ public class Mapping {
     }
 
     private Object convertType(String value) {
-        if (nullValues.contains(value)) return null;
+        if (nullValues.contains(value) || (StringUtils.isBlank(value) && !type.equals(Meta.Types.STRING))) {
+            return null;
+        }
         if (type == Meta.Types.STRING) return value;
 
         final Supplier<ZoneId> timezone = () -> ZoneId.of((String) optionalData.getOrDefault("timezone", apocConfig().getString(db_temporal_timezone.name())));

--- a/core/src/test/java/apoc/export/csv/ImportCsvTest.java
+++ b/core/src/test/java/apoc/export/csv/ImportCsvTest.java
@@ -102,6 +102,11 @@ public class ImportCsvTest {
                             ":ID(MultiType-ID)|date1:datetime{timezone:Europe/Stockholm}|date2:datetime|foo:string|joined:date|active:boolean|points:int\n" +
                             "1|2018-05-10T10:30|2018-05-10T12:30|Joe Soap|2017-05-05|true|10\n" +
                             "2|2018-05-10T10:30[Europe/Berlin]|2018-05-10T12:30[Europe/Berlin]|Jane Doe|2017-08-21|true|15\n"),
+                    new AbstractMap.SimpleEntry<>("emptyDate",
+                            "id:ID,:LABEL,str:STRING,int:INT,date:DATE\n" +
+                                    "1,Lab,hello,1,2020-01-01\n" +
+                                    "2,Lab,world,2,2020-01-01\n" +
+                                    "3,Lab,,,\n"),
                     new AbstractMap.SimpleEntry<>("relMultiTypes",
                             ":START_ID(MultiType-ID)|:END_ID(MultiType-ID)|prop1:IGNORE|prop2:time{timezone:+02:00}[]|foo:int|time:duration[]|baz:localdatetime[]|bar:localtime[]\n" +
                             "1|2|a|15:30|1|P14DT16H12M|2020-01-01T00:00:00|11:00:00\n" +
@@ -429,6 +434,26 @@ public class ImportCsvTest {
 
         Assert.assertEquals("John Jane", TestUtil.singleResultFirstColumn(db, "MATCH (p1:Person)-[:FRIENDS_WITH]->(p2:Person) RETURN p1.name + ' ' + p2.name AS pair ORDER BY pair"));
         Assert.assertEquals("Jane John", TestUtil.singleResultFirstColumn(db, "MATCH (p1:Person)-[:KNOWS]->(p2:Person) RETURN p1.name + ' ' + p2.name AS pair ORDER BY pair"));
+    }
+
+    @Test
+    public void testEmptyDate() {
+        TestUtil.testCall(db, "CALL apoc.import.csv([{fileName: 'file:/emptyDate.csv', labels: ['Entity']}], [], {date: {nullValues: ['']}})",
+                r -> assertEquals(3L, r.get("nodes")));
+
+        TestUtil.testResult(db, "MATCH (node:Entity:Lab) RETURN node ORDER BY node.id", r -> {
+            final Node firstNode = (Node) r.next().get("node");
+            final Map<String, Object> expectedFirstNode = Map.of("date", LocalDate.of(2020, 1, 1),
+                    "int", 1L, "id", "1", "str", "hello");
+            assertEquals(expectedFirstNode, firstNode.getAllProperties());
+            final Node secondNode = (Node) r.next().get("node");
+            final Map<String, Object> expectedSecondNode = Map.of("date", LocalDate.of(2020, 1, 1), 
+                    "int", 2L, "id", "2", "str", "world");
+            assertEquals(expectedSecondNode, secondNode.getAllProperties());
+            final Node thirdNode = (Node) r.next().get("node");
+            assertEquals(Map.of("str", "", "id", "3"), thirdNode.getAllProperties());
+            assertFalse(r.hasNext());
+        });
     }
     
     @Test

